src/main/java/org/littleshoot/proxy/extras/CertificateSniffingMitmManager.java

```diff
@@ -0,0 +1,108 @@
+package org.littleshoot.proxy.extras;
+
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLPeerUnverifiedException;
+import javax.net.ssl.SSLSession;
+
+import org.littleshoot.proxy.MitmManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link MitmManager} that tries to find the common name and subject
+ * alternative names from the upstream certificate to create a dynamic
+ * certificate with it.
+ * 
+ * @deprecated There is a problem with long subject alternate name lists in the
+ *             upstream certificate. See the comments.
+ */
+@Deprecated
```

Comment:
> Just curious... why mark this new class @Deprecated

I've modified the API and this class shows the reason for. Use 
HostNameMitmManager instead of CertificateSniffingMitmManager please.

First: it's simpler to impersonate by host name. Second: long lists of 
subject alternative names are truncated somewhere. A bug? But: it's 
important to store the *complete* upstream cert to present it to the 
user anyway.

Please give me both, the host name and the SSL session in the manager.